### PR TITLE
Clang

### PR DESCRIPTION
**Pull request by clero**
Add clang support to parameter-framework project.
Clang comes in Android with its standard library which supports c++11.

### CODE REVIEW

**Comment by clero** [comment]
@dawagner @krocard please review

File: .travis.yml [hunk 1]
```diff
@@ -28,6 +26,14 @@ install:
...
     - wget --directory-prefix $PREFIX/include
               https://raw.github.com/philsquared/Catch/master/single_include/catch.hpp
+    # Install recent swig.
+    # Avoid make install check because documentation generation will fail as yodl2man command is
+    # missing and the corresponding paquet is not available in travis whitelist.
+    - ( mkdir build_swig && cd build_swig &&
+          wget https://github.com/swig/swig/archive/rel-2.0.12.tar.gz &&
+          tar -xf rel-2.0.12.tar.gz && cd swig-rel-2.0.12/ && ./autogen.sh &&
+          ./configure --prefix=$PREFIX && make && make install ;
+          echo "Dropping documentation generation error" )
```
Comment (clero) [inline .travis.yml]: try to avoid documentation generation

File: .travis.yml [hunk 2]
```diff
@@ -55,6 +61,7 @@ after_success:
           --exclude "build_debug/bindings/python"
           --exclude "build_debug/CMakeFiles"
           --exclude "build"
+          --exclude "build_swig"
```
Comment (clero) [inline .travis.yml]: build in the home directory to avoid the exclude

**Comment by dawagner** [comment]
Obsoleted by #133.

**Pull request closed without merging**